src/list.js

```diff
@@ -7,49 +7,65 @@ export default Primitive.extend('List', {
 
   get length() { return this.value.length; },
 
-  fill() {
-    return this.replace(this.value.slice().fill(...arguments));
-  },
-
-  pop() {
-    return this.slice(0, -1);
+  prototype: {
+    forEach: { value() { return this.value.forEach(...arguments); }},
+    valueOf: { value() { return this.value.valueOf(); }}
   },
```

Comment:
I'm guessing we can get rid of the whole `prototype` hash since we're now storing the transitions inside its own hash so it will never conflict.